paymentsheet/src/main/java/com/stripe/android/paymentsheet/ui/BaseSheetActivity.kt

```diff
@@ -378,8 +387,27 @@ internal abstract class BaseSheetActivity<ResultType> : AppCompatActivity() {
         bottomSheet.layoutParams = params
     }
 
+    protected inline fun <T> StateFlow<T>.collectInActivity(crossinline transform: (T) -> Unit) {
+        lifecycleScope.launch {
+            repeatOnLifecycle(Lifecycle.State.STARTED) {
```

Comment:
is this actually required for all calls? This would be re-trigered when the activity/Fragment recovers. If you are sending network request here, it might get sent mutiple times.

Reply:
You can test it by setting not keep activity and backgrounding the app and bring it back to front

Reply:
I tested this out, but I am not getting re-triggers of the internals of the collect. What I found was that the network calls were contained in the ViewModel lifecycle scope and gets cancelled correctly when you background the app (killing the coroutine since DKA is enabled).

Reply:
Yes, the coroutine will be cancelled if it's not finished yet when app is killed. 
But regardless it's cancelled or not, the same request will be send again if the app is brought to front end again(with or without DKA is enabled), it should be fine if the API request is idempotent, but just want to make sure this is what's expected.

E.g with the following snippet collecting an `val intFlow = MutableStateFlow<Int>()`, i tried to background the app when `sendSomeRequestAndHandleResponse` is not finished(between //1 and //2), then bring the app back to front(//3) , in either case, another request is made at //3, just want to make sure this is something that's expect to happen here. 

* with DNK off:
```
15:15:04.055  D  new value 23!, sending a request // 1
15:15:09.060  D  new value 23!, request done // 2
15:15:11.045  D  new value 23!, sending a request // 3
15:15:16.047  D  new value 23!, request done // 4
```

* with DNK on:
```
15:15:44.937  D  new value 23!, sending a request // 1
15:15:47.038  D  cancelled with value 23 // 2
15:15:49.783  D  new value 23!, sending a request // 3
15:15:54.788  D  new value 23!, request done // 4
```

```
override fun onCreate(savedInstanceState: Bundle?) {
        viewModel.intFlow.collectInActivity {
            lifecycleScope.launch {
                try {
                    Log.d("test", "new value $it! sending a request")
                    viewModel.sendSomeRequestAndHandleResponse()
                    // or just use delay(5000) to mimic a long running task
                    Log.d("test", "new value $it! request finished")
                } catch (e: CancellationException) {
                    Log.d("test", "cancelled with value $it")
                }
            }
        }
}
```

I think `repeatOnLifecycle` is suitable for UI as explained in [this](https://medium.com/androiddevelopers/repeatonlifecycle-api-design-story-8670d1a7d333) post, but we might be cautious when use it for something else